File: mobile/minibilge_mobile/lib/features/auth/models/user_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'parent_profile_dto.dart';

part 'user_dto.freezed.dart';
part 'user_dto.g.dart';

@freezed
class UserDto with _$UserDto {
  const factory UserDto({
    @JsonKey(name: 'Id') required String id,
    @JsonKey(name: 'Email') required String email,
    @JsonKey(name: 'Role') required String role,
    @JsonKey(name: 'CanUseOnlineSpeech') @Default(false) bool canUseOnlineSpeech,
    @JsonKey(name: 'IsTeacher') @Default(false) bool isTeacher,
    @JsonKey(name: 'ParentProfile') ParentProfileDto? parentProfile,
  }) = _UserDto;

  factory UserDto.fromJson(Map<String, dynamic> json) =>
      _$UserDtoFromJson(json);
}

File: backend/MiniBilge.Application/DTOs/Auth/UserDto.cs
using MiniBilge.Application.DTOs.Profile;

namespace MiniBilge.Application.DTOs.Auth;

public class UserDto
{
    public Guid Id { get; set; }
    public string Email { get; set; } = string.Empty;
    public string Role { get; set; } = string.Empty;
    public bool CanUseOnlineSpeech { get; set; }
    public string ExperienceMode { get; set; } = string.Empty;
    public ParentProfileDto? ParentProfile { get; set; }
}
